packages/cli/src/modules/insights/repositories/insights-by-period.repository.ts

```diff
@@ -1,11 +1,83 @@
-import { Service } from '@n8n/di';
+import { GlobalConfig } from '@n8n/config';
+import { Container, Service } from '@n8n/di';
 import { DataSource, Repository } from '@n8n/typeorm';
+import { z } from 'zod';
+
+import { sql } from '@/utils/sql';
 
 import { InsightsByPeriod } from '../entities/insights-by-period';
+const dbType = Container.get(GlobalConfig).database.type;
+
+const summaryParser = z
+	.object({
+		period: z.enum(['previous', 'current']),
+		// TODO: extract to abstract-entity
+		type: z.union([z.literal(0), z.literal(1), z.literal(2), z.literal(3)]),
```

Comment:
Are we leaving TODO's in the code? Should we raise an issue in Linear to do the work?